src/Aspire.Hosting.Azure.AIFoundry/AzureAIFoundryExtensions.cs

```diff
@@ -0,0 +1,360 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Aspire.Hosting.ApplicationModel;
+using Aspire.Hosting.Azure;
+using Aspire.Hosting.Azure.AIFoundry;
+using Aspire.Hosting.Eventing;
+using Azure.Provisioning;
+using Azure.Provisioning.CognitiveServices;
+using Azure.Provisioning.Expressions;
+using Azure.Provisioning.Resources;
+using Microsoft.AI.Foundry.Local;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Diagnostics.HealthChecks;
+using Microsoft.Extensions.Logging;
+using static Azure.Provisioning.Expressions.BicepFunction;
+
+namespace Aspire.Hosting;
+
+/// <summary>
+/// Provides extension methods for adding the Azure AI Foundry resources to the application model.
+/// </summary>
+public static class AzureAIFoundryExtensions
+{
+    /// <summary>
+    /// Adds an Azure OpenAI resource to the application model.
+    /// </summary>
+    /// <param name="builder">The <see cref="IDistributedApplicationBuilder"/>.</param>
+    /// <param name="name">The name of the resource. This name will be used as the connection string name when referenced in a dependency.</param>
+    /// <returns>A reference to the <see cref="IResourceBuilder{T}"/>.</returns>
+    public static IResourceBuilder<AzureAIFoundryResource> AddAzureAIFoundry(this IDistributedApplicationBuilder builder, [ResourceName] string name)
+    {
+        builder.AddAzureProvisioning();
+
+        var resource = new AzureAIFoundryResource(name, ConfigureInfrastructure);
+        return builder.AddResource(resource)
+                      .WithManifestPublishingCallback(resource.WriteToManifest);
```

Comment:
You don't need this call because it is done in the base AzureBicepResource ctor.